contributors/design-proposals/server-get.md

```diff
@@ -0,0 +1,170 @@
+
+Today, all clients must reproduce the tabular and describe output implemented in `kubectl` to perform simple lists
+of objects. This logic in many cases is non-trivial and condenses multiple fields into succinct output. It also requires
+that every client provide rendering logic for every possible type, including those provided by API aggregation or third
+party resources which may not be known at compile time.
+
+This proposal covers moving `get` and `describe` to the server to reduce total work for all clients and centralize these
+core display options for better reuse and extension.
+
+
+## Background
+
+`kubectl get` is a simple tabular representation of one or more instances of a particular resource type. It is the primary
+listing mechanism and so must be implemented for each type. Today, we have a single generic implementation for unrecognized
+types that attempts to load information from the `metadata` field (assuming the object follows the metav1 Kubernetes API
+schema). `get` supports a `wide` mode that includes additional columns. Users can add additional columns for labels via a
+flag. Headers corresponding to the columns are optionally displayed.
+
+`kubectl describe` shows a textual representation of individual objects that describes individual fields as subsequent
+lines and uses indendation and nested tables to convey deeper structure on the resource (such as events for a pod or
+each container). It sometimes retrieves related objects like events, pods for a replication controller, or autoscalers
+for a deployment. It supports no significant flags.
+
+The implementation of both is modeled as a registered function that takes an object or list of objects and outputs
+semi-structured text.
+
+## Goals
+
+* Make it easy for a simple client to get a list of resources for a web UI or CLI output
+* Support all existing options, leave open the door for future extension and experimentation
+* Allow new API extensions and third party resources to be implemented server side, removing the need to version
+  schemes for retrieving data from the server
+* Keep implementation of `get` and `describe` simple
+* Ease internationalization of `get` and `describe` output for all clients
+
+## Non-Goals
+
+* Deep customization of the returned output by the client
+
+
+## Specification of server-side `get`
+
+The server would return a `TableList` object (working-name) that contains metadata for columns and one or more
+rows composed of cells for each column.  Some additional data may be relevant for each row and returned by the
+server. Since every object should have a `TableList` representation, treat this as part of content negotiation
+as described in [the alternative representations of objects proposal](alternate-api-representations.md).
+
+Example request:
+
+```
+$ curl https://localhost:8443/api/v1/pods -H "Accept: application/json+vnd.kubernetes.as+meta.k8s.io+v1alpha1+TableList"
+{
+    "kind": "TableList",
```

Comment:
I think the only difference here is that subsequent pages could omit the headers.